api-docs: Document incoming webhook URL specification and modal to generate these URLs in web-app. (#27066)
zulip/zulip: api_docs/incoming-webhooks-overview.md

  testing with live data from the service you're integrating and can help you
  spot why something isn't working or if the service is using custom HTTP
  headers.

## URLs

The base URL for an incoming webhook integration bot is

> `{{ api_url }}/v1/external/integration_name?api_key=abcdefgh` where
> `integration_name` is the name of the specific webhook integration and
> `api_key` is the API key of the bot created by the user for the
> integration.

Reviewer: This is a bit confusing because `integration_name` and `api_key` are implicitly given two different conventions for how to replace things in the example: the reader is expected to replace the literal text "integration_name", but the value that comes after the literal text "api_key". I think both of those are pretty readable conventions, but the juxtaposition is odd.

Here's one way to use a single convention for both:

```suggestion
`{{ api_url }}/v1/external/INTEGRATION_NAME?api_key=API_KEY` where
`INTEGRATION_NAME` is the name of the specific webhook integration and
`API_KEY` is the API key of the bot created by the user for the
integration.
```

The list of existing webhook integrations can be found in

> `zerver/lib/integrations.py` (see `WEBHOOK_INTEGRATIONS`) or by browsing

Reviewer: ```suggestion
`zerver/lib/integrations.py` (at `WEBHOOK_INTEGRATIONS`) or by browsing
```

or perhaps "see `WEBHOOK_INTEGRATIONS` there"

My first reading was that "see `WEBHOOK_INTEGRATIONS`" was a reference meant to further elaborate on the meaning of "`zerver/lib/integrations.py`", so this is just to disambiguate from that.

the [Integrations documentation](/integrations).

> To get a bot's API key, see the [API keys](/api/api-keys) documentation.
> 
> To send incoming webhook notifications to a specific stream, pass a
> stream name ([URL-encoded][url-encoder]) or stream ID as a `stream`
> parameter in the URL. If a stream parameter is not passed in the URL,
> then the integration will send direct messages to the bot's owner. And
> to send notifications to a specific topic in the specified stream, then
> pass a [URL-encoded][url-encoder] topic as a `topic` parameter in the
> URL.
> 
> Note that some incoming webhook integrations also support passing events
> in the URL. Details about supported events for an integration can be
> found on its [integration documentation](/integrations) page.

Reviewer: I think this might benefit from a bit more of a tabular or list format, rather than pure prose, especially for the `stream` and `topic` part. Something like this:

```suggestion
Parameters accepted in the URL include:

* `api_key`: Required.
  The API key of the bot created by the user for the integration.
  To get a bot's API key, see the [API keys](/api/api-keys) documentation.

* `stream`: The stream for …. By default the integration will send
  direct messages ….

* `topic`: …

* `only_events`, `exclude_events`: Some incoming webhook integrations
  support these parameters to filter which events will trigger a
  notification.  For details, see the integration's
  [integration documentation](/integrations) page.
```
